### PR TITLE
bugfix(tests): using JUnit's ResourceLock annotation for tests that touch…

### PR DESCRIPTION
**Pull request by lprimak**
… global SecurityManager

Following this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [GitHub issue](https://github.com/apache/shiro/issues) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a GitHub issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[#XXX] - Fixes bug in SessionManager`,
       where you replace `#XXX` with the appropriate GitHub issue. Best practice
       is to use the GitHub issue title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] add `fixes #XXX` if merging the PR should close a related issue.
 - [x] Run `mvn verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`. 
 - [x] Committers: Make sure a milestone is set on the PR
 
Trivial changes like typos do not require a GitHub issue (javadoc, comments...). 
In this case, just format the pull request title like `[DOC] - Add javadoc in SessionManager`.
 
If this is your first contribution, you have to read the [Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md)

If your pull request is about ~20 lines of code you don't need to sign an [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) 
if you are unsure please ask on the developers list.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -274,6 +274,7 @@
                     <artifactId>maven-surefire-plugin</artifactId>
                     <configuration>
                         <argLine>@{jacocoAgent} @{surefire.argLine}</argLine>
+                        <parallel>all</parallel>
```
Comment (bmarwell) [inline pom.xml:277]: Doesn't work with junit Jupiter. Instead, use their properties in argLine or resource property files.

The latter will be picked up by IDEs

**Review by bmarwell — CHANGES_REQUESTED** [review]

**Review by bmarwell — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -274,17 +274,25 @@
...
+                            </configurationParameters>
+                        </properties>
                     </configuration>
                 </plugin>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-failsafe-plugin</artifactId>
                     <configuration>
                         <argLine>@{jacocoAgent} @{failsafe.argLine}</argLine>
+                        <parallel>classes</parallel>
```
[the remark this answers is not in the document]
Comment (lprimak) [reply pom.xml:295]: Probably not gonna happen for failsafe... getting integration tests to run in parallel is a *lot* more difficult

File: pom.xml [hunk 3]
```diff
@@ -278,6 +278,12 @@
                         <enableProcessChecker>native</enableProcessChecker>
                         <trimStackTrace>false</trimStackTrace>
                         <redirectTestOutputToFile>true</redirectTestOutputToFile>
+                        <properties>
+                            <configurationParameters>
+                                junit.jupiter.execution.parallel.enabled = true
+                                junit.jupiter.execution.parallel.mode.default = concurrent
```
Comment (bmarwell) [inline pom.xml:284]: Have you tried concurrent methods as well?

Comment (bmarwell) [inline pom.xml:284]:
```suggestion
                                junit.jupiter.execution.parallel.mode.default = concurrent
                                junit.jupiter.execution.parallel.mode.classes.default = concurrent
```

File: pom.xml [hunk 4]
```diff
@@ -278,6 +278,12 @@
                         <enableProcessChecker>native</enableProcessChecker>
                         <trimStackTrace>false</trimStackTrace>
                         <redirectTestOutputToFile>true</redirectTestOutputToFile>
+                        <properties>
+                            <configurationParameters>
```
Comment (bmarwell) [inline pom.xml:282]: `configurationParameters` does not exist: https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html

Either create a junit-platform.properties in every `src/test/resources` folder, OR use surefire's argLine.

The first approach will have the advantage that IntelliJ and others will pick it up. The second is DRY-compatible (but that may actually be a disadvantage)

**Review by bmarwell — DISMISSED** [review]

File: pom.xml (hunk 4 above)
Comment (lprimak) [reply pom.xml:282]: Tests run in parallel. Taken from here: https://rieckpil.de/parallelize-only-java-unit-tests-with-junit-5-and-maven/

File: pom.xml (hunk 3 above)
Comment (lprimak) [reply pom.xml:284]: I don’t think your addition is actually necessary. That is on by default and can be overridden to be disabled

Comment (lprimak) [reply pom.xml:284]: I have done a quick test and `junit.jupiter.execution.parallel.mode.classes.default = concurrent` has no effect if `junit.jupiter.execution.parallel.mode.default = concurrent` is already present, which it is :)

File: pom.xml (hunk 1 above)
Comment (lprimak) [reply pom.xml]: done

File: pom.xml [hunk 5]
```diff
@@ -274,17 +274,25 @@
                     <artifactId>maven-surefire-plugin</artifactId>
                     <configuration>
                         <argLine>@{jacocoAgent} @{surefire.argLine}</argLine>
+                        <parallel>all</parallel>
```
[the remark this answers is not in the document]
Comment (lprimak) [reply pom.xml]: done

File: pom.xml (hunk 4 above)
Comment (bmarwell) [reply pom.xml:282]: Found the docs:
https://maven.apache.org/surefire/maven-surefire-plugin/examples/junit-platform.html

File: config/ogdl/src/test/groovy/org/apache/shiro/config/ogdl/ReflectionBuilderTest.groovy [hunk 6]
```diff
@@ -36,6 +37,7 @@ import static org.hamcrest.Matchers.*
 /**
  * Unit tests for the {@link org.apache.shiro.config.ogdl.ReflectionBuilder} implementation.
  */
+@Isolated
```
Comment (bdemers) [inline config/ogdl/src/test/groovy/org/apache/shiro/config/ogdl/ReflectionBuilderTest.groovy:40]: nit: add reason for needing to be isolated (we can possibly fix it in the future)

File: core/src/test/groovy/org/apache/shiro/authc/credential/DefaultPasswordServiceTest.groovy [hunk 7]
```diff
@@ -38,6 +39,7 @@ import static org.junit.jupiter.api.Assertions.*
  *
  * @since 1.2
  */
+@Isolated
```
Comment (bdemers) [inline core/src/test/groovy/org/apache/shiro/authc/credential/DefaultPasswordServiceTest.groovy:42]: nit: add reason

File: core/src/test/java/org/apache/shiro/authc/AbstractAuthenticatorTest.java [hunk 8]
```diff
@@ -40,6 +41,7 @@
 /**
  * @since 0.1
  */
+@Isolated
```
Comment (bdemers) [inline core/src/test/java/org/apache/shiro/authc/AbstractAuthenticatorTest.java:44]: nit: add reason

File: support/ehcache/src/test/java/org/apache/shiro/cache/ehcache/EhCacheManagerTest.java [hunk 9]
```diff
@@ -37,6 +38,7 @@
  *
  * @since May 11, 2010 12:41:38 PM
  */
+@Isolated
```
Comment (bdemers) [inline support/ehcache/src/test/java/org/apache/shiro/cache/ehcache/EhCacheManagerTest.java:41]: nit: add reason

File: tools/hasher/src/test/java/org/apache/shiro/tools/hasher/HasherTest.java [hunk 10]
```diff
@@ -42,6 +43,7 @@
 /**
  * @since 2.0
  */
+@Isolated
```
Comment (bdemers) [inline tools/hasher/src/test/java/org/apache/shiro/tools/hasher/HasherTest.java:46]:
```suggestion
@Isolated("Uses System Input")
```

File: web/src/test/groovy/org/apache/shiro/web/config/IniFilterChainResolverFactoryTest.groovy [hunk 11]
```diff
@@ -41,6 +42,7 @@ import static org.hamcrest.MatcherAssert.assertThat
  *
  * @since 1.2
  */
+@Isolated
```
Comment (bdemers) [inline web/src/test/groovy/org/apache/shiro/web/config/IniFilterChainResolverFactoryTest.groovy:45]: nit: add reason

My guess is that the INI / reflection builder _might_ be sys prop related?

File: web/src/test/groovy/org/apache/shiro/web/filter/InvalidRequestFilterTest.groovy [hunk 12]
```diff
@@ -29,6 +30,7 @@ import static org.easymock.EasyMock.mock
 import static org.easymock.EasyMock.replay
 import static org.hamcrest.MatcherAssert.assertThat
 
+@Isolated
```
Comment (bdemers) [inline web/src/test/groovy/org/apache/shiro/web/filter/InvalidRequestFilterTest.groovy:33]:
```suggestion
@Isolated("Uses System Properties")
```

File: web/src/test/groovy/org/apache/shiro/web/servlet/AbstractShiroFilterTest.groovy [hunk 13]
```diff
@@ -31,6 +33,7 @@ import static org.junit.jupiter.api.Assertions.*
 /**
  * Unit tests for the {@link AbstractShiroFilter} implementation.
  */
+@Isolated
```
Comment (bdemers) [inline web/src/test/groovy/org/apache/shiro/web/servlet/AbstractShiroFilterTest.groovy:36]:
```suggestion
@Isolated("Uses System Properties and Static Security Manager")
```

File: web/src/test/groovy/org/apache/shiro/web/util/WebUtilsTest.groovy [hunk 14]
```diff
@@ -31,7 +32,8 @@ import static org.junit.jupiter.api.Assertions.*
 /**
  * Tests for {@link WebUtils}.
  */
-public class WebUtilsTest {
+@Isolated
```
Comment (bdemers) [inline web/src/test/groovy/org/apache/shiro/web/util/WebUtilsTest.groovy:35]:
```suggestion
@Isolated("Uses System Properties")
```

File: web/src/test/java/org/apache/shiro/web/mgt/DefaultWebSecurityManagerTest.java [hunk 15]
```diff
@@ -56,6 +57,7 @@
 /**
  * @since 0.9
  */
+@Isolated
```
Comment (bdemers) [inline web/src/test/java/org/apache/shiro/web/mgt/DefaultWebSecurityManagerTest.java:60]: nit: add reason

**Review by bdemers — COMMENTED** [review]
+0.9
This is great!
I left a few comments about adding adding a `value` to the `Isolated` annotation. suggestions are in there too, but they were added via the GH UI...

This is a non-blocking review 😉

**Review by bmarwell — APPROVED** [review]

File: config/ogdl/src/test/groovy/org/apache/shiro/config/ogdl/ReflectionBuilderTest.groovy (hunk 6 above)
Comment (lprimak) [reply config/ogdl/src/test/groovy/org/apache/shiro/config/ogdl/ReflectionBuilderTest.groovy:40]: haha Dunno.. cause it failed :)

**Pull request merged**